src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,108 @@
-import React from 'react';
+/* eslint-disable max-len */
+import React, { useEffect, useState } from 'react';
+import {
+  addComment,
+  deleteComment,
+  getPostComments,
+  getPostDetails,
+} from '../../api/api';
+import { Comment } from '../../types/comment';
+import { Post } from '../../types/post';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
+type Props = {
+  selectedPostId: number,
+};
 
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
+export const PostDetails: React.FC<Props> = ({ selectedPostId }) => {
+  const [post, setPost] = useState<Post>();
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentsShown, setIsCommentsShown] = useState<boolean>(true);
```

Comment:
you don't need generics for primitives types, they are set automatically (if state won't have any other type)